### PR TITLE
suggestion service written

### PR DESCRIPTION
**Pull request by In-vincible**
Only service written not ready to merge.

### CODE REVIEW

**Comment by In-vincible** [comment]
@hemantjadon should I import the service and start, using the service in feed?

File: src/app/shared/services/suggest/suggest.service.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import 'rxjs/add/operator/catch';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/observable/throw';
+
+import { SuggestResponse } from '../../../models/api-response';
+
+@Injectable()
+export class SuggestService {
+	private static apiUrl: URL = new URL('http://api.loklak.org/api/suggest.json');
+	//private static maximum_records_fetch: string = '10';
```
Comment (hemantjadon) [inline src/app/shared/services/suggest/suggest.service.ts]: Remove this comment

File: src/app/shared/services/suggest/suggest.service.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	private static minified_results: string = 'true';
```
Comment (hemantjadon) [inline src/app/shared/services/suggest/suggest.service.ts]: Make this a boolean (true) and use .toString() method later when setting query params

Comment (In-vincible) [inline src/app/shared/services/suggest/suggest.service.ts]: I tried to change this with boolean, but it is throwing this error "Argument of type 'boolean' is not assignable to parameter of type 'string'.", we are receiving it in string form.

Comment (In-vincible) [inline src/app/shared/services/suggest/suggest.service.ts]: Build is failing due to error in provider of jsonp I think this has to do something with component which I have not yet added

Comment (hemantjadon) [inline src/app/shared/services/suggest/suggest.service.ts]: > .toString() method later when setting query params

Comment (In-vincible) [inline src/app/shared/services/suggest/suggest.service.ts]: sorry got it now done.

**Comment by In-vincible** [comment]
@hemantjadon I am writing @ngrx/effects for SuggestService in /effects/api-suggest.effects.ts, I hope this is the right way to go, keep suggesting :)

**Comment by In-vincible** [comment]
I think I can use  actionTypeCheck  from '../shared/utils', I have to write new reducers in app/reducers,
I have to add actions in actions, than comes effects, it's the way to go according to me, please correct me where I am wrong.

**Comment by hemantjadon** [comment]
Yes its the correct way of doing it,

* First write actions, `suggest.ts`
   * SUGGEST
   * SUGGEST_COMPLETE_SUCCESS
   * SUGGEST_COMPLETE_FAIL

* Then define the State `suggest.ts`
   * query: SuggestQuery
   * loading: boolean

* Then write reducers and selectors. in `suggest.ts` like done in `search.ts`

* Then define the reducers and state for `api-suggest.ts` 

* The state will include: `api-suggest.ts`
   * metadata: ApiSuggestMetadata
   * entities: ApiSuggestResult[]
   * valid: boolean

**Comment by hemantjadon** [comment]
@In-vincible move the service to the new location, ie. `app/services` instead of `app/shared/services` please update your fork also as i have made few changes in directory structure

**Comment by In-vincible** [comment]
@hemantjadon I am planning to write components for suggestion, should I start in a new folder src/app/suggest  or I can fit in home or somewhere...and ReloactionAfterQuery in query what does it do?

File: src/app/actions/suggest.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+/**
+ * For each action type in an action group, make a simple
+ * enum object for all of this group's action types.
+ *
+ * The 'actionTypeCheck' utility function coerces strings into string
+ * literal types and runs a simple check to guarantee all
+ * action types in the application are unique.
+ */
+export const ActionTypes = {
+	SUGGEST: actionTypeCheck('[Api] Search'),
```
Comment (hemantjadon) [inline src/app/actions/suggest.ts]: Wrong duplicate action types. Write action types as [API] Suggest etc

Comment (In-vincible) [inline src/app/actions/suggest.ts]: yeah I was about to make these changes :+1:

**Comment by hemantjadon** [comment]
@In-vincible  any updates in this PR?

**Comment by In-vincible** [comment]
I have written effects for the suggestion, now I think I am ready to inject it in module and add eventlisteners for suggestions.

**Comment by mariobehling** [comment]
@hemantjadon @In-vincible Is this PR still ongoing?

**Comment by hemantjadon** [comment]
@mariobehling Yes this PR's work is okay, If @In-vincible don't want to carry this further I would suggest merging this then let someone else take over.

**Comment by mariobehling** [comment]
@In-vincible Thank you.

@hemantjadon Please open an issue for next steps. Cheers.

**Pull request merged**